test/widgets/lightbox_test.dart

```diff
@@ -558,4 +601,51 @@ void main() {
       check(platform.position).equals(position);
     });
   });
-}
+
+  group('LightboxHero', () {
+    testWidgets('no hero animation occurs between different message list pages for same image', (tester) async {
+      final channel = eg.stream(streamId: eg.defaultStreamMessageStreamId);
+      final message = eg.streamMessage(stream: channel,
+        contentMarkdown: ContentExample.spoilerHeaderHasImage.html, topic: 'test topic');
+      await setupMessageListPage(tester, narrow: const CombinedFeedNarrow(),
+        streams: [channel], messages: [message]);
+
+      connection.prepare(json:
+        eg.newestGetMessagesResult(foundOldest: true, messages: [message]).toJson());
+
+      final imageFinder = find.byType(RealmContentNetworkImage).first;
+      final initialImageRect = tester.getRect(imageFinder);
+
+      await tester.tap(find.descendant(
+        of: find.byType(StreamMessageRecipientHeader),
+        matching: find.text('test topic')));
+      await tester.pump();
+      await tester.pump(const Duration(milliseconds: 150));
+
+      final imageInTransition = tester.getRect(imageFinder);
+      check(imageInTransition.top).equals(initialImageRect.top);
+      check(imageInTransition.left).equals(initialImageRect.left);
+      await tester.pumpAndSettle();
```

Comment:
Is this supposed to be mid transition time? While this works, I feel that it can miss bugs if the duration somehow changes.

We can have a while-loop that checks `tester.hasRunningAnimations`, which pumps repeatedly with a fixed duration, and repeats the check:

```suggestion

      int timeElapsed = 0;
      const interval = 50;
      while (timeElapsed < interval || tester.hasRunningAnimations) {
        final imageInTransition = tester.getRect(imageFinder);
        check(imageInTransition.top).equals(initialImageRect.top);
        check(imageInTransition.left).equals(initialImageRect.left);
        await tester.pump(const Duration(milliseconds: interval));
        timeElapsed += interval;
      }
      check(timeElapsed).isGreaterOrEqual(interval);
```

Reply:
Just saw the [CZO comment](https://chat.zulip.org/#narrow/channel/516-mobile-dev-help/topic/Hero.20animation/near/2095334)[#mobile-dev-help > Hero animation @ 💬](https://chat.zulip.org/#narrow/channel/516-mobile-dev-help/topic/Hero.20animation/near/2095334):

> In order to have that benefit, it's important to keep the two test cases similar to each other as far as possible. It should be easy for the reader to convince themself that the two tests are checking the same thing, and just expecting opposite outcomes.

I agree that having these two tests similar to each other does mitigate the concern of one of them not working properly. So it's probably also fine to leave them as-is. Either way, we should comment that the duration is specifically chosen such that we are in the middle of a hero animation (if there is one).